dpath/common.py

```diff
@@ -0,0 +1,28 @@
+import sys
+if sys.version_info < (3,):
+    def u(x):
+        x = x if isinstance(x, basestring) else str(x)
+        return try_decode(x)
+else:
+    def u(x):
+        return str(x)
+
+def try_decode(obj):
+    """
+    Tries to decode a basestring (python 2/3) into a unicode string.
+    Returns the OBJ if it fails or if it isn't a basestring. Otherwise returns the unicode version of the
+    base string.
+    """
+    try:
+        if isinstance(obj, basestring):
+            return obj.decode('utf-8')
```

Comment:
This assumption is going to wind up biting us at some point. Unicode and utf8 are not the same thing. Strings might be perfectly valid unicode, but if they're utf16, this will blow up. This will work in the vast majority of cases, but there will certainly be a corner case in the future that brings us right back here. I'd like to see something less "hulk smash" re: detecting the encoding, preferrably something that relies on user input.